bqplot/plotting_widgets.py

```diff
@@ -0,0 +1,207 @@
+    """
+
+    data = DataFrame().tag(dtype=None)
+    data_range = List([0, 1]).tag(sync=True)
+    band_type = Enum(
+        ["circle", "polygon"], default_value="circle", allow_none=True
+    ).tag(sync=True)
+    colors = List(default_value=CATEGORY10).tag(sync=True)
+    num_bands = Int(default_value=5).tag(sync=True)
+    fill = Bool(default_value=False).tag(sync=True)
+
+    def __init__(self, **kwargs):
+        super(Radar, self).__init__(**kwargs)
+        self.scales = {"x": LinearScale(), "y": LinearScale()}
+        # set some defaults for the figure
+        self.layout = Layout(min_width="600px", min_height="600px")
+        self.max_aspect_ratio = 1
+        self.preserve_aspect = True
+
+        # marks for the radar figure
+
+        # spokes (straight lines going away from the center)
+        self.spokes = Lines(
+            scales=self.scales, colors=["#ccc"], stroke_width=0.5
+        )
+
+        # bands
+        self.bands = Lines(
+            colors=["#ccc"], scales=self.scales, stroke_width=0.5
+        )
+
+        # loops of the radar
+        self.loops = Lines(
+            scales=self.scales,
+            display_legend=True,
+            colors=self.colors,
+            stroke_width=2,
+            fill="inside" if self.fill else "none",
+            marker="circle",
+            marker_size=50,
+        )
+
+        self.band_labels = Label(
+            scales=self.scales,
+            default_size=12,
+            font_weight="normal",
+            apply_clip=False,
+            colors=["#ccc"],
+            align="middle",
+        )
+
+        self.spoke_labels = Label(
+            scales=self.scales,
+            default_size=14,
+            font_weight="bold",
+            apply_clip=False,
+            colors=["#ccc"],
+            align="middle",
+        )
+
+        self.marks = [
+            self.spokes,
+            self.bands,
+            self.loops,
+            self.band_labels,
+            self.spoke_labels,
+        ]
+
+        # handlers for data updates
+        self.observe(self.update_data, "data")
+        self.observe(self.update_bands, ["band_type", "num_bands"])
+        self.observe(self.update_fill, "fill")
+
+        self.loops.on_legend_click(self.on_legend_click)
+        self.loops.on_background_click(self.reset)
+
+        self.update_bands(None)
+        self.update_data(None)
+
+    def update_bands(self, *args):
+        band_data = np.linspace(
+            self.data_range[0], self.data_range[1], self.num_bands + 1
+        )
+        self.scaled_band_data = (
+            (band_data - self.data_range[0])
+            / (self.data_range[1] - self.data_range[0])
+        )[:, np.newaxis]
+
+        n = len(self.data.index)
+
+        if self.band_type == "circle":
+            t = np.linspace(0, 2 * np.pi, 1000)
+            band_data_x, band_data_y = (
+                self.scaled_band_data * np.cos(t),
+                self.scaled_band_data * np.sin(t),
+            )
```

Comment:
It would be nice to have a `Circle` primitive (similar to the `Lines` one), so we don't need to create a circle based on a thousand segments.

This would be more efficient as it would mean sending 3 float values (center + radius) instead of 2000 float values to the front-end.

Just adding this as a note, I don't think we should block this PR waiting for such a primitive.